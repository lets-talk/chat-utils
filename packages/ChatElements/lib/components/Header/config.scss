@import '../../common/styles/colors';
@import '../../common/styles/variables';

$widgetBox-Header-background: $color-background;
$widgetBox-Header-color: $color-white;
$widgetBox-Header-fontsize: $font-size-normal;
$widgetBox-Header-text-align: center;
$widgetBox-Header-title-fontsize: $font-size-big;
$widgetBox-Header-subtitle-fontsize: $font-size-small;
