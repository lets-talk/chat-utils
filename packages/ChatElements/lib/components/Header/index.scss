@import "./config.scss";

@mixin letstalk-header {
  flex-shrink: 0;
  position: relative;
}

@mixin letstalk-title {
  font-size: $widgetBox-Header-title-fontsize;
}

@mixin letstalk-minimize-button {
  background-color: transparent;
  border: 0;
  display: block;
  position: absolute;
  left: 10px;
  top: 15px;
  width: 40px;
}

@mixin letstalk-menu-button {
  background-color: transparent;
  border: 0;
  display: block;
  position: absolute;
  right: 10px;
  top: 15px;
  width: 40px;
}

.letstalk-header {
  @include letstalk-header;
  @include themify($themes) {
    color: themed('color-text');
    background-color: themed('color-background-primary');
  }
  border-radius: 10px 10px 0 0;
  color: white;
  display: flex;
  flex-direction: column;
  text-align: $widgetBox-Header-text-align;
  padding: 15px 20px;
}

.letstalk-title {
  @include letstalk-title;
  font-size: $widgetBox-Header-title-fontsize;
  font-weight: normal;
  margin: 0 40px;
  padding: 5px 10px;
}

.letstalk-subtitle {
  font-size: $widgetBox-Header-subtitle-fontsize;
}

.letstalk-minimize-button {
  @include letstalk-minimize-button;
}

.letstalk-menu-button {
  @include letstalk-menu-button;
}
