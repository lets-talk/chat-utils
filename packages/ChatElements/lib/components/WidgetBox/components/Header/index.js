import React from 'react';
import PropTypes from 'prop-types';

import MdKeyboardArrowDown from 'react-icons/lib/md/keyboard-arrow-down';
import MdMenu from 'react-icons/lib/md/menu';
import Avatar from '../../../Avatar';

import './index.scss';

const Header = ({
  title, subtitle, avatar, toggleChat, openMenu, showMinimizeButton, showMenuButton,
}) =>
  (
    <div className="letstalk-header">
      {avatar && <Avatar src={avatar} size="medium" withStatus status="online" />}
      <h4 className="letstalk-title">{title}</h4>
      <span className="letstalk-subtitle">{subtitle}</span>
      {
        showMinimizeButton &&
        <button className="letstalk-minimize-button" onClick={toggleChat}>
          <MdKeyboardArrowDown size={20} color="white" />
        </button>
      }
      {
        showMenuButton &&
        <button className="letstalk-menu-button" onClick={openMenu}>
          <MdMenu size={20} color="white" />
        </button>
      }
    </div>
  );

Header.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string,
  avatar: PropTypes.string,
  toggleChat: PropTypes.func,
  openMenu: PropTypes.func,
  showMinimizeButton: PropTypes.bool,
  showMenuButton: PropTypes.bool,
};

export default Header;
