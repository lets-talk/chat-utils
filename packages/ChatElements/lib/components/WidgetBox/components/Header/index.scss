@import '../../../../common/styles/variables.css';
@import "../common.scss";

.letstalk-header {
  background-color: var(--widgetBox-Header-background);
  border-radius: 10px 10px 0 0;
  color: white;
  display: flex;
  flex-direction: column;
  text-align: var(--widgetBox-Header-text-align);
  padding: 15px 20px;
}

.letstalk-title {
  font-size: var(--widgetBox-Header-title-fontsize);
  font-weight: normal;
  margin: 0 40px;
  padding: 5px 10px;
}

.letstalk-subtitle {
  font-size: var(--widgetBox-Header-subtitle-fontsize);
}

.letstalk-minimize-button {
  display: none;
}

@media screen and (max-width: 800px) {
  .letstalk-header {
    @include letstalk-header;
  }

  .letstalk-title {
    @include letstalk-title;
  }

  .letstalk-minimize-button {
    @include letstalk-minimize-button;
  }

  .letstalk-minimize {
    @include letstalk-minimize;
  }

  .letstalk-menu-button {
    @include letstalk-menu-button;
  }

  .letstalk-menu {
    @include letstalk-menu;
  }
}
