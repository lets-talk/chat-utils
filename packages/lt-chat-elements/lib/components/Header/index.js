import React from 'react';
import PropTypes from 'prop-types';

import MdKeyboardArrowDown from 'react-icons/lib/md/keyboard-arrow-down';
import styled from 'styled-components';
import { Flex, Box } from 'grid-styled';
import MdMenu from 'react-icons/lib/md/menu';
import Avatar from '../Avatar';
import { headerMenuButton, themeColor, textColor } from '../../utils/style';

const StyledHeaderContainer = styled.div`
  background-color: ${(props) => themeColor(props.theme, 'accent', 'base')};
  color: ${(props) => textColor(props.theme, 'dark', 'primary')};
  border-radius: 10px 10px 0 0;
  text-align: center;
  padding: 15px 20px;

  button {
    color: ${(props) => textColor(props.theme, 'dark', 'primary')};
    ${headerMenuButton('40px')};
  }

  h4 {
    margin: 0;
    font-size: ${(props) => props.theme.typography.classes.body.fontSize};
    line-height: ${(props) => props.theme.typography.classes.body.lineHeight};
    font-weight: ${(props) => props.theme.typography.weights.fontWeightMedium};
  }

  span {
    color: ${(props) => textColor(props.theme, 'dark', 'secondary')};
    font-size: ${(props) => props.theme.typography.classes.caption.fontSize};
    line-height: ${(props) => props.theme.typography.classes.caption.lineHeight};
    font-weight: ${(props) => props.theme.typography.weights.fontWeightThin};
  }
`;

const StyledAvatarContainer = styled.div`
  margin-bottom: 10px;
`;

const Header = ({
  className, title, subtitle, person, toggleChat, openMenu, showMinimizeButton, showMenuButton,
}) => {
  const { avatar = false, status = '' } = person;
  return (
    <StyledHeaderContainer className={className}>
      <Flex alignItems="center">
        <Box flex="0 0 auto" mb="auto" width="40px">
          {showMenuButton &&
            <button onClick={openMenu}>
              <MdMenu size={20} />
            </button>
          }
        </Box>

        <Box flex="1 1 auto">
          {avatar &&
            <StyledAvatarContainer>
              <Avatar
                src={avatar}
                size="medium"
                withStatus={status.length > 0}
                status={status}
              />
            </StyledAvatarContainer>
          }
          {title && <h4>{title}</h4>}
          {subtitle && <span>{subtitle}</span>}
        </Box>

        <Box flex="0 0 auto" mb="auto" width="40px">
          {showMinimizeButton &&
            <button onClick={toggleChat}>
              <MdKeyboardArrowDown size={20} />
            </button>
          }
        </Box>
      </Flex>
    </StyledHeaderContainer>
  );
};

Header.propTypes = {
  className: PropTypes.string,
  title: PropTypes.string,
  subtitle: PropTypes.string,
  person: PropTypes.shape({
    avatar: PropTypes.string,
    status: PropTypes.string,
    email: PropTypes.string,
    type: PropTypes.string,
  }),
  toggleChat: PropTypes.func,
  openMenu: PropTypes.func,
  showMinimizeButton: PropTypes.bool,
  showMenuButton: PropTypes.bool,
};

Header.defaultProps = {
  className: 'LT-Header-Container',
  title: '',
  subtitle: '',
  person: {},
  toggleChat: null,
  openMenu: null,
  showMinimizeButton: true,
  showMenuButton: true,
};

Header.displayName = 'Header';

export default Header;
