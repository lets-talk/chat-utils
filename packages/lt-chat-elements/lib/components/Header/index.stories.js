import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, boolean, select, text, object } from '@storybook/addon-knobs';

// Component to show on storybook
import Header from '.';
import { WrapWithTheme, defaultTheme, themeOptions } from '../../utils/stories';
import withTests from '../../test-utils/storiesWithTest';

const stories = storiesOf('Header', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests('Header'));

const leftButtons = [
  {
    onClick: () => action('onLeftButtonClick'),
    icon: <i>Icon Left</i>,
  },
];

const rightButtons = [
  {
    onClick: () => action('onRightButtonClick'),
    icon: <b>Any node</b>,
  },
];

const agent = {
  avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
  email: '',
  type: 'Agent',
  status: 'online',
};

stories.addWithInfo(
  'Header',
  'This is the basic usage with the Header',
  () => (
    <div
      style={{
        position: 'fixed', bottom: 0, right: '10px', width: 'calc(100% - 20px)',
      }}
    >
      <WrapWithTheme
        themeName={select('Theme', themeOptions, defaultTheme)}
      >
        <Header
          title={text('Title', 'Ayuda en LÃ­nea')}
          subtitle={text('Subtitle', 'Cambio de clave')}
          toggleChat={action('toggleChat')}
          openMenu={action('openMenu')}
          showMenuButton={boolean('Show Menu Button', false)}
          showMinimizeButton={boolean('Show Minimize Button', false)}
          person={object('Person', agent)}
          leftButtons={leftButtons}
          rightButtons={rightButtons}
        />
      </WrapWithTheme>
    </div>
  )
);
