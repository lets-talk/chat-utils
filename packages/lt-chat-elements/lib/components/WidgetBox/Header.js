import React from 'react';
import PropTypes from 'prop-types';

import styled from 'styled-components';
import { headerMenuButton, themeColor, textColor } from '../../utils/style';

const StyledHeaderContainer = styled.div`
  flex: 0 0 auto;
  flex-basis: ${(props) => `${props.height}px`};

  background-color: ${(props) => themeColor(props.theme, 'accent', 'base')};
  color: ${(props) => textColor(props.theme, 'dark', 'primary')};
  border-radius: 10px 10px 0 0;
  text-align: center;
  padding: 15px 20px;

  button {
    color: ${(props) => textColor(props.theme, 'dark', 'primary')};
    ${headerMenuButton('40px')};
  }

  h4 {
    margin: 0;
    font-size: ${(props) => props.theme.typography.classes.body.fontSize};
    line-height: ${(props) => props.theme.typography.classes.body.lineHeight};
    font-weight: ${(props) => props.theme.typography.weights.fontWeightMedium};
  }

  span {
    color: ${(props) => textColor(props.theme, 'dark', 'secondary')};
    font-size: ${(props) => props.theme.typography.classes.caption.fontSize};
    line-height: ${(props) => props.theme.typography.classes.caption.lineHeight};
    font-weight: ${(props) => props.theme.typography.weights.fontWeightThin};
  }
`;
StyledHeaderContainer.displayName = 'StyledHeaderContainer';

const Header = (props) => {
  const { children, className, height } = props;
  return (
    <StyledHeaderContainer className={className} height={height}>
      {children}
    </StyledHeaderContainer>
  );
};

Header.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  height: PropTypes.number,
};

Header.defaultProps = {
  className: 'LT-Header-Container',
  height: 50,
};

Header.displayName = 'Header';

export default Header;
