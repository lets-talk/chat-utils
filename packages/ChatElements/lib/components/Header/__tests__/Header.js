import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from '../index';

const sleep = (time) => new Promise((resolve) => setTimeout(resolve, time));

describe('Header component', () => {
  it('should render with minimal props', () => {
    const component = shallow(<Header  />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with data', () => {
    const props = {
      title: 'Header Title',
      subtitle: 'Header subtitle',
      avatar: '',
      avatarStatus: 'online',
      toggleChat: () => null,
      openMenu: () => null,
      showMinimizeButton: true,
      showMenuButton: true,
    };

    const component = shallow(<Header {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  describe('Header with openMenu and toggleChat functions handlers', () => {
    it('Test toggleChat function handler is called when click on proper button', () => {
      const mockToggleChat = jest.fn();
      const fakeEventObject = { preventDefault() {}, stopPropagation() {} };

      const props = {
        title: 'Header Title',
        subtitle: 'Header subtitle',
        avatar: '',
        avatarStatus: 'online',
        openMenu: () => null,
        showMinimizeButton: true,
        showMenuButton: true,
      };

      const button = shallow((<Header toggleChat={mockToggleChat} {...props} />));
      button.find('.letstalk-minimize-button').simulate('click', fakeEventObject);
      expect(mockToggleChat.mock.calls.length).toEqual(1);
    });
  });

  it('Test openMenu function handler is being called when clicked on proper button', () => {
    const mockOpenMenu = jest.fn();
    const fakeEventObject = { preventDefault() {}, stopPropagation() {} };

    const props = {
      title: 'Header Title',
      subtitle: 'Header subtitle',
      avatar: '',
      avatarStatus: 'online',
      toggleChat: () => null,
      showMinimizeButton: true,
      showMenuButton: true,
    };

    const button = shallow((<Header openMenu={mockOpenMenu} {...props} />));
    button.find('.letstalk-menu-button').simulate('click', fakeEventObject);
    expect(mockOpenMenu.mock.calls.length).toEqual(1);
  });
});
