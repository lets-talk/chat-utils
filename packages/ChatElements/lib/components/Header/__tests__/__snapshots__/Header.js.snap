// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render with data 1`] = `
<div
  className="letstalk-header"
>
  <button
    className="letstalk-minimize-button"
    onClick={[Function]}
  >
    <MdKeyboardArrowDown
      color="white"
      size={20}
    />
  </button>
  <button
    className="letstalk-menu-button"
    onClick={[Function]}
  >
    <MdMenu
      color="white"
      size={20}
    />
  </button>
  <h4
    className="letstalk-title"
  >
    Header Title
  </h4>
  <span
    className="letstalk-subtitle"
  >
    Header subtitle
  </span>
</div>
`;

exports[`Header component should render with minimal props 1`] = `
<div
  className="letstalk-header"
>
  <button
    className="letstalk-minimize-button"
    onClick={null}
  >
    <MdKeyboardArrowDown
      color="white"
      size={20}
    />
  </button>
  <button
    className="letstalk-menu-button"
    onClick={null}
  >
    <MdMenu
      color="white"
      size={20}
    />
  </button>
  <h4
    className="letstalk-title"
  />
  <span
    className="letstalk-subtitle"
  />
</div>
`;
